pipe.go

```diff
@@ -223,9 +224,36 @@ func DialPipe(path string, timeout *time.Duration) (net.Conn, error) {
 // DialPipeContext attempts to connect to a named pipe by `path` until `ctx`
 // cancellation or timeout.
 func DialPipeContext(ctx context.Context, path string) (net.Conn, error) {
+	conn, err := DialPipeContextAccess(ctx, path, syscall.GENERIC_READ|syscall.GENERIC_WRITE)
+	if err == context.DeadlineExceeded {
```

Comment:
Also for context based methods we dont want to convert to `ErrTimeout`. It is correct to return `ctx.Err()` which is `context.DeadlineExceeded` from `tryDialPipe`. We only convert in the `DialPipe` for legacy reasons.